File: assignments_nilesh_sir/java_assignments/Assignment_6/MetTours/PremiumTours.java
package MetTours;

public class PremiumTours {

    /*
     * Here we calculate PerDayRent depending on the no of Persons.
     */

    @LuxuryTax(value = 10)
    public double getPerDayRentForCommon(int days, int noPersons) {
        return noPersons * 1200;
    }

    @LuxuryTax(value = 10)
    public double getPerDayRentForSeniors(int days, int noPersons) {
        return getPerDayRentForCommon(days, noPersons) - 100;
    }

    @LuxuryTax(value = 10)
    public double getPerDayRentForWoman(int days, int noPersons) {
        return getPerDayRentForCommon(days, noPersons) - 200;
    }

}

File: Dotnet/07-05-24/MetTours/PremiumTours.cs
namespace MetTours;

public class PremiumTours
{
    static int discount = 0;

    public double GetPerDayRentForCommon(int Days, int NoOfPersons)
    {
        discount = (Days > 6 ? Days * 100 : 0) + (NoOfPersons >= 4 ? NoOfPersons * 100 : 0);
        return (NoOfPersons * 1200) - discount;
    }

    public double GetPerDayRentForSeniors(int Days, int NoOfPersons)
    {
        return GetPerDayRentForCommon(Days, NoOfPersons) - (NoOfPersons * 100);
    }

    public double GetPerDayRentForWomen(int Days, int NoOfPersons)
    {
        return GetPerDayRentForCommon(Days, NoOfPersons) - (NoOfPersons * 200);
    }
}